src/scatter_buildblock/upsample_and_fit_scatter_estimate.cxx

```diff
@@ -53,12 +54,22 @@ upsample_and_fit_scatter_estimate(ProjData& scaled_scatter_proj_data,
 {
   shared_ptr<ProjDataInfo> 
     interpolated_direct_scatter_proj_data_info_sptr(emission_proj_data.get_proj_data_info_sptr()->clone());
-  interpolated_direct_scatter_proj_data_info_sptr->reduce_segment_range(0,0);
 
   info("upsample_and_fit_scatter_estimate: Interpolating scatter estimate to size of emission data");
   ProjDataInMemory interpolated_direct_scatter(emission_proj_data.get_exam_info_sptr(),
 					       interpolated_direct_scatter_proj_data_info_sptr);        
-  interpolate_projdata(interpolated_direct_scatter, scatter_proj_data, spline_type, remove_interleaving);
+  
+  if(emission_proj_data.get_proj_data_info_sptr()->get_scanner_sptr()->get_scanner_geometry()=="BlocksOnCylindrical")
+  {
+      bool remove_interleaving_block=false;
+      interpolated_direct_scatter_proj_data_info_sptr->reduce_segment_range(0,0);
+      interpolate_projdata(interpolated_direct_scatter, scatter_proj_data, spline_type, remove_interleaving_block);
+  }
+  else
+  {
+      interpolated_direct_scatter_proj_data_info_sptr->reduce_segment_range(0,0);
+      interpolate_projdata(interpolated_direct_scatter, scatter_proj_data, spline_type, remove_interleaving);
+  }
```

Comment:
can't this be done simpler? do we need a different `bool`? I guess we could force it ` false` in `set-up`

Reply:
in set-up?

Reply:
```suggestion
  bool actual_remove_interleaving = remove_interleaving;
  if (remove_interleaving && emission_proj_data.get_proj_data_info_sptr()->get_scanner_sptr()->get_scanner_geometry()!="Cylindrical")
  {
     warning("upsample_and_fit_scatter_estimate: forcing remove_interleaving to false as non-cylindrical projdata");
     actual_remove_interleaving = false;
  }
  interpolated_direct_scatter_proj_data_info_sptr->reduce_segment_range(0,0);
  interpolate_projdata(interpolated_direct_scatter, scatter_proj_data, spline_type, actual_remove_interleaving);
```

Reply:
However, the line
```
interpolated_direct_scatter_proj_data_info_sptr->reduce_segment_range(0,0);
```
should happen BEFORE the construction of `interpolated_direct_scatter` (as it was before). Now it will have no effect

Reply:
so shall I move it there?